homeassistant/components/light/hue.py

```diff
@@ -60,251 +46,142 @@
     'Color temperature light': SUPPORT_HUE_COLOR_TEMP
     }
 
-CONF_ALLOW_IN_EMULATED_HUE = "allow_in_emulated_hue"
-DEFAULT_ALLOW_IN_EMULATED_HUE = True
-
-CONF_ALLOW_HUE_GROUPS = "allow_hue_groups"
-DEFAULT_ALLOW_HUE_GROUPS = True
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Optional(CONF_HOST): cv.string,
-    vol.Optional(CONF_ALLOW_UNREACHABLE): cv.boolean,
-    vol.Optional(CONF_FILENAME): cv.string,
-    vol.Optional(CONF_ALLOW_IN_EMULATED_HUE): cv.boolean,
-    vol.Optional(CONF_ALLOW_HUE_GROUPS,
-                 default=DEFAULT_ALLOW_HUE_GROUPS): cv.boolean,
-})
-
-ATTR_GROUP_NAME = "group_name"
-ATTR_SCENE_NAME = "scene_name"
-SCENE_SCHEMA = vol.Schema({
-    vol.Required(ATTR_GROUP_NAME): cv.string,
-    vol.Required(ATTR_SCENE_NAME): cv.string,
-})
-
 ATTR_IS_HUE_GROUP = "is_hue_group"
 
-CONFIG_INSTRUCTIONS = """
-Press the button on the bridge to register Philips Hue with Home Assistant.
 
-![Location of button on bridge](/static/images/config_philips_hue.jpg)
-"""
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Hue lights."""
+    setup_data(hass)
 
+    for bridge in hass.data[hue.DOMAIN].values():
+        update_lights(hass, bridge, add_devices)
 
-def _find_host_from_config(hass, filename=PHUE_CONFIG_FILE):
-    """Attempt to detect host based on existing configuration."""
-    path = hass.config.path(filename)
+    return True
 
-    if not os.path.isfile(path):
-        return None
 
-    try:
-        with open(path) as inp:
-            return next(json.loads(''.join(inp)).keys().__iter__())
-    except (ValueError, AttributeError, StopIteration):
-        # ValueError if can't parse as JSON
-        # AttributeError if JSON value is not a dict
-        # StopIteration if no keys
-        return None
+def setup_data(hass):
+    """Initialize internal data. Useful from tests."""
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {'lights': {}, 'lightgroups': {}}
```

Comment:
let's replace `lights` and `lightgroups` with `DATA_LIGHTS` and `DATA_LIGHTGROUPS` constants